runtime/sema/checker.go

```diff
@@ -341,51 +366,50 @@ func (checker *Checker) CheckProgram(program *ast.Program) {
 	}
 }
 
-func (checker *Checker) checkTopLevelDeclarationValidity(declarations []ast.Declaration) {
+func (checker *Checker) checkTopLevelDeclarationsValidity(declarations []ast.Declaration) {
 	validTopLevelDeclarationsHandler := checker.Config.ValidTopLevelDeclarationsHandler
 
 	if validTopLevelDeclarationsHandler == nil {
 		return
 	}
 
-	validDeclarationKinds := map[common.DeclarationKind]bool{}
```

Comment:
We were previously allocating a map for each check

Reply:
Nice optimization 👌